pkg/config/config.go

```diff
@@ -20,21 +20,24 @@ type Module struct {
 	ACL  []string `toml:"acl"`
 }
 
+func NewModule(name string, path string, acl []string) Module {
+	return Module{
+		Name: name,
+		Path: path,
+		ACL:  acl,
+	}
+}
+
 type Config struct {
-	Listeners []Listener        `toml:"listener"`
-	Modules   []Module          `toml:"module"`
-	ModuleMap map[string]Module `toml:"-"`
+	Listeners []Listener `toml:"listener"`
+	Modules   []Module   `toml:"module"`
 }
 
 func FromString(input string) (*Config, error) {
 	var cfg Config
 	if _, err := toml.Decode(input, &cfg); err != nil {
 		return nil, err
 	}
-	cfg.ModuleMap = make(map[string]Module)
-	for _, m := range cfg.Modules {
-		cfg.ModuleMap[m.Name] = m
-	}
```

Comment:
One nice place of improvement is here: no need to compute the module map. One less place where data can be out of sync (think somebody later changes the `Modules` fields but does not realize `ModuleMap` is actually the authoritative one.

Same for ModuleMap key (being the name). If someone changes the `Name` in the `Module` structure, they've to remember to change the map key too.